docs/Reference-for-template.json.md

```diff
@@ -492,11 +492,12 @@ See [the article](Binding-and-project-context-evaluation.md#bind-symbols).
 |Name|Description|Default|Mandatory|
 |---|---|---|---|
 |`sourceName`| The text in the source content to replace with the name the user specifies. The value to be replaced with can be given using the `-n` `--name` options while running a template. The template engine will look for any occurrence of the `sourceName` present in the config file and replace it in file names and file contents. If no name is specified by the host, the current directory is used. The value of the `sourceName` is available in built-in `name` symbol and can be used as the source for creating other symbols and condition expressions. Due to implicit forms applied to `sourceName` it is important to select the replacement that doesn't cause the conflicts. See more details about `sourceName` in [the article](Naming-and-default-value-forms.md)||no|
+|`preferDefaultName`| Boolean value that decides which name will be used if no `--name` is specified during creation. If `false` it will use the fallback name, if `true` it will use the template's `defaultName`. If no `defaultName` is specified, the template creation will not succeed. | If not specified, `false` is used. | no
```

Comment:
```suggestion
|`preferDefaultName`| Boolean value that decides which name will be used if no `--name` is specified during creation. If `false` it will use the fallback name (output folder name), if `true` it will use the template's `defaultName`. If no `defaultName` is specified and `preferDefaultName` is set to `true`, the template creation will not succeed. | If not specified, `false` is used. | no
```

what is the error in this case: no `defaultName` is specified and `preferDefaultName` is set to `true`?

Reply:
It will result in a `MissingMandatoryParam` status if there is no fallback name specified. If there is a fallback name, then it will be that name. I realize that this is not clear in the docs, so I will adjust it accordingly.

Reply:
It would be better if it results in `TemplateIssueDetected`  status https://github.com/dotnet/templating/blob/c0f7124770e08c67b86197cb655c7eed7410096c/src/Microsoft.TemplateEngine.Edge/Template/CreationResultStatus.cs#L26 
as it is an authoring error with correct error message. The user cannot fix it - it's authoring error. 

@JanKrivanek please correct me if the status i mentioned  is wrong.
We should also added the check for this in https://github.com/dotnet/templating/issues/2623

Reply:
That's correct - `TemplateIssueDetected` would be better fit in this case.

Btw. it seems we are missing the `defaultName` in this doc. @maridematte - can you please add the `defaultName` into the `Template Definition` section (around the line 40)? It might be as well helpful to crerate html anchor there and link it from `defaultName` mention on this line.